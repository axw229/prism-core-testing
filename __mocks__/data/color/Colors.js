// @flow
import cloneDeep from 'lodash/cloneDeep'
import sample from 'lodash/sample'

import type { ColorSetPayload, Color } from '../../../src/shared/types/Colors'

export const Colors: ColorSetPayload = {
  "Red": [
    [
      {
        "colorNumber": "6561",
        "coordinatingColors": {
          "coord2ColorId": 1926,
          "coord1ColorId": 2250,
          "whiteColorId": 2681
        },
        "description": [
          "Subdued",
          "Muted",
          "Brilliant",
          "Radiant"
        ],
        "id": "2248",
        "isExterior": true,
        "isInterior": true,
        "name": "Teaberry",
        "rgb": 15454683,
        "lrv": 68.551,
        "colorFamilyNames": [
          "Red"
        ],
        "brandKey": "SW",
        "red": 235,
        "green": 209,
        "blue": 219,
        "hue": 0.935897435897436,
        "saturation": 0.39393939393939387,
        "lightness": 0.8705882352941177,
        "hex": "#ebd1db",
        "isDark": false,
        "cssrgb": "rgb(235,209,219)",
        "storeStripLocator": "102-C1",
        "similarColors": [
          "2255",
          "2262",
          "2269",
          "2662",
          "2535",
          "2276",
          "2658",
          "1974",
          "2543",
          "2531"
        ]
      },
      {
        "colorNumber": "6568",
        "coordinatingColors": {
          "coord2ColorId": 11244,
          "coord1ColorId": 2687,
          "whiteColorId": 2681
        },
        "description": [
          "Subdued",
          "Muted",
          "Brilliant",
          "Radiant"
        ],
        "id": "2255",
        "isExterior": true,
        "isInterior": true,
        "name": "Lighthearted Pink",
        "rgb": 15586781,
        "lrv": 70.758,
        "brandedCollectionIds": [
          31797,
          31791
        ],
        "brandedCollectionNames": [
          "Pottery Barn Fall/Winter 2018",
          "Pottery Barn Fall/Winter 2018"
        ],
        "colorFamilyNames": [
          "Red"
        ],
        "brandKey": "SW",
        "red": 237,
        "green": 213,
        "blue": 221,
        "hue": 0.9444444444444445,
        "saturation": 0.3999999999999999,
        "lightness": 0.8823529411764706,
        "hex": "#edd5dd",
        "isDark": false,
        "cssrgb": "rgb(237,213,221)",
        "storeStripLocator": "103-C1",
        "similarColors": [
          "2248",
          "2262",
          "2269",
          "2276",
          "1974",
          "1981",
          "2658",
          "1995",
          "2662",
          "1988"
        ]
      }
    ]
  ]
}

export function getColors (): ColorSetPayload {
  return cloneDeep(Colors)
}

export function getColor (): Color | void {
  const colors = getColors()
  const fam1 = Object.keys(colors)[0]
  if (colors[fam1]) {
    return sample(colors[fam1][0])
  }
}
