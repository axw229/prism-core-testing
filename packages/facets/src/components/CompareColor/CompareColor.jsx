// @flow
import React from 'react'
import { fullColorNumber, getContrastYIQ } from '../../../src/shared/helpers/ColorUtils'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import 'src/providers/fontawesome/fontawesome'
import type { Color } from '../../shared/types/Colors.js.flow'
import type { FlatScene, FlatVariant } from '../../shared/types/Scene'
import WithConfigurationContext, { type ConfigurationContextType } from '../../contexts/ConfigurationContext/WithConfigurationContext'
import './CompareColor.scss'
import { FormattedMessage } from 'react-intl'
import SingleTintableSceneView from '../SingleTintableSceneView/SingleTintableSceneView'
import { createMiniColorFromColor } from '../SingleTintableSceneView/util'

const baseClass = 'color__comparison'
const containerClass = `${baseClass}__container`
const containerHeaderClass = `${baseClass}__container__header`
const containerHeaderButtonClass = `${baseClass}__container__header__button`
const wrapperClass = `${baseClass}__wrapper`
const prevBtnWrapperClass = `${baseClass}__prev-btn__wrapper`
const buttonsClass = `${baseClass}__buttons`
const buttonsVisibleClass = `${baseClass}__buttons--visible`
const nextBtnWrapperClass = `${baseClass}__next-btn__wrapper`
const backgroundColorClass = `${baseClass}__background__color`
const queueWrapperClass = `${baseClass}__queue__wrapper`

const defaultViewItem = 3
const offset = 1

type CompareColorProps = {
  colors: Color[],
  config: ConfigurationContextType,
  toggleCompareColor: Function,
  colorIds: string[],
  scenesCollection: FlatScene[],
  variantsCollection: FlatVariant[],
  selectedSceneUid: string,
  selectedVariantName: string
}

type CompareColorState = {
  colors: Color[],
  colorIds: string[],
  renderColors: Color[],
  curr: number,
  isHideNextButton: boolean,
  isHidePrevButton: boolean
}
export class CompareColor extends React.Component<CompareColorProps, CompareColorState> {
  state = {
    colors: this.props.colors,
    colorIds: this.props.colorIds,
    renderColors: [],
    curr: 0,
    isHidePrevButton: true,
    isHideNextButton: true,
    selectedSceneUid: this.props.selectedSceneUid,
    selectedVariantName: this.props.selectedVariantName
  }

  static getDerivedStateFromProps (props: CompareColorProps, state: CompareColorState) {
    if (props.colors !== state.colors) {
      return {
        colors: props.colors
      }
    }
    if (props.colorIds !== state.colorIds) {
      if (props.colorIds.length === state.colors.length - 1) {
        props.toggleCompareColor()
        return null
      }
      let currPointer = state.curr
      // when remove compared color id from color paltete
      if (props.colorIds.length > state.colorIds.length) {
        if (state.colors.length - props.colorIds.length <= defaultViewItem || currPointer === 0) {
          currPointer = 0
        } else {
          currPointer = currPointer - offset
        }
      }
      // when add compared color id from color pallate
      if (props.colorIds.length < state.colorIds.length) {
        if (state.colors.length - props.colorIds.length - 1 <= defaultViewItem) {
          currPointer = 0
        } else {
          currPointer = currPointer + 1
        }
      }
      return {
        colorIds: props.colorIds,
        curr: currPointer
      }
    }
    return null
  }

  componentDidUpdate (prevProps, prevState) {
    if (prevProps.selectedSceneUid !== this.props.selectedSceneUid ||
        prevProps.selectedVariantName !== this.props.selectedVariantName) {
      // eslint-disable-next-line react/no-did-update-set-state
      this.setState({
        selectedSceneUid: this.props.selectedSceneUid,
        selectedVariantName: this.props.selectedVariantName
      })
    }
  }

  renderContent = (curr: number, colors: Colors[], colorIds: string, colorInfoClass: string, colorNumOnBottom: boolean, selectedSceneUid: string, selectedVariantName: string, scenesCollection: FlatScene[], variantsCollection: FlatVariant[]) => {
    const renderColors = colors.filter((color: Color) => {
      return !colorIds.includes(color.id)
    })
    return renderColors && renderColors.map((color, index) => {
      const selectedVariant = variantsCollection.find(variant => variant.sceneUid === selectedSceneUid && variant.variantName === selectedVariantName)
      const selectedScene = scenesCollection.find(scene => scene.uid === selectedSceneUid)
      const colorDataTemplate = selectedVariant.surfaces.map(surface => null)
      const fillFirstSurfaceColor = (template, color) => {
        const miniColor = createMiniColorFromColor(color)
        return template.map((placeholder, i) => {
          return i ? null : miniColor
        })
      }
      const colorsForView = fillFirstSurfaceColor(colorDataTemplate, color)
      return (
        <div key={index} style={{ backgroundColor: color.hex, transform: `translateX(-${curr * 100}%)` }} className={`${queueWrapperClass}`}>
          <div className={`${backgroundColorClass}`} />
          <SingleTintableSceneView
            surfaceColorsFromParents={colorsForView}
            selectedSceneUid={selectedSceneUid}
            scenesCollection={[selectedScene]}
            variantsCollection={[selectedVariant]} />
          <div style={{ color: getContrastYIQ(color.hex) }} className={`${colorInfoClass}${colorNumOnBottom ? '__name-number' : ''}`} >
            <span className={`${colorInfoClass}__number`}>{fullColorNumber(color.brandKey, color.colorNumber)}</span>
            <span className={`${colorInfoClass}__name`}>{color.name}</span>
          </div>
        </div>
      )
    })
  }

  isShowSlideButton = () => {
    const { curr, colors, colorIds } = this.state
    const itemsCount = colors.length - colorIds.length
    const buttonVisibility = {
      isHidePrevButton: true,
      isHideNextButton: true
    }
    if (curr === 0) {
      buttonVisibility.isHidePrevButton = true
    } else {
      buttonVisibility.isHidePrevButton = false
    }

    if (itemsCount > defaultViewItem && curr < itemsCount - defaultViewItem) {
      buttonVisibility.isHideNextButton = false
    }

    if (curr === itemsCount - defaultViewItem - offset / 2) {
      buttonVisibility.isHideNextButton = true
    }
    return buttonVisibility
  }

  handlePrev = () => {
    const { colors, curr, colorIds } = this.state
    const itemsCount = colors.length - colorIds.length
    if (itemsCount > defaultViewItem) {
      if (curr === (offset / 2)) {
        this.setState({ curr: curr - offset / 2 })
      } else {
        this.setState({ curr: curr - offset })
      }
    }
  }

  handleNext = () => {
    const { colors, curr, colorIds } = this.state
    const itemsCount = colors.length - colorIds.length
    if (itemsCount > defaultViewItem) {
      if (curr === itemsCount - defaultViewItem - 1) {
        this.setState({ curr: curr + offset / 2 })
      } else {
        this.setState({ curr: curr + offset })
      }
    }
  }

  render () {
    const { curr, colors, colorIds, selectedSceneUid, selectedVariantName } = this.state
    const { config, variantsCollection, scenesCollection } = this.props
    const { colorWall: { colorSwatch = {} }, cvw = {} } = config
    const { colorNumOnBottom = false, houseShaped = false } = colorSwatch
    const { closeBtn = {} } = cvw
    const { showArrow: closeBtnShowArrow = true, text: closeBtnText = <FormattedMessage id='CLOSE' /> } = closeBtn
    const colorInfoClass = houseShaped ? `${baseClass}-house-shaped` : `${baseClass}__color__info`
    const content = this.renderContent(curr, colors, colorIds, colorInfoClass, colorNumOnBottom, selectedSceneUid, selectedVariantName, scenesCollection, variantsCollection)
    const { isHidePrevButton, isHideNextButton } = this.isShowSlideButton()

    return (
        <div className={`${containerClass}`}>
          <div className={`${containerHeaderClass}`}>
            <span><FormattedMessage id='COMPARE_COLORS' /></span>
            <button className={`${containerHeaderButtonClass}`} onClick={this.props.toggleCompareColor}> {closeBtnText ?? <FormattedMessage id='CLOSE' />}{closeBtnShowArrow && <FontAwesomeIcon className={`${containerHeaderClass}--icon`} icon={['fa', 'chevron-up']} />}</button>
          </div>
          <div className={`${wrapperClass}`} >
            <div className={`${prevBtnWrapperClass}`}>
              {
                <button className={`${buttonsClass} ${isHidePrevButton ? `${buttonsVisibleClass}` : ''}`} onClick={this.handlePrev}>
                  <FontAwesomeIcon icon={['fa', 'chevron-left']} />
                </button>}
            </div>
            {content}
            <div className={`${nextBtnWrapperClass}`}>
              {
                <button className={`${buttonsClass} ${isHideNextButton ? `${buttonsVisibleClass}` : ''}`} onClick={this.handleNext}>
                  <FontAwesomeIcon icon={['fa', 'chevron-right']} />
                </button>
              }
            </div>
          </div>
        </div>
    )
  }
}

export {
  containerClass,
  containerHeaderClass,
  containerHeaderButtonClass,
  wrapperClass,
  prevBtnWrapperClass,
  buttonsClass,
  buttonsVisibleClass,
  nextBtnWrapperClass,
  backgroundColorClass,
  queueWrapperClass
}

export default WithConfigurationContext(CompareColor)
