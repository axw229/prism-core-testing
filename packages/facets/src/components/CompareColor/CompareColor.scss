.color__comparison {
  &__container {
    font-family: var(getVarName('typography.titleFontFamily'));
  }
  &__container__header {
    display: flex;
    justify-content: space-between;
    font-size: 1.125rem;
    padding: 1em 0;
    position: relative;
    background-color: var(getThemeColor('white'));
    & > span {
      margin: 0 auto;
      align-self: center;
      text-transform: var(getVarName('typography.titleTextTransform'));
    }
    &__button {
      text-transform: var(getVarName('typography.buttonTextTransform'));
      font-family: var(getVarName('typography.titleFontFamily'));
      font-weight: var(getVarName('typography.buttonFontWeight'));
      border: 1px solid var(getThemeColor('buttonBorder'));
      background-color: var(getThemeColor('buttonBgColor'));
      color: var(getThemeColor('buttonColor'));
      text-align: center;
      fill: currentColor;
      padding: 0.75em 1.75em;
      outline: none;
      position: absolute;
      top:50%;
      right:1rem;
      transform: translateY(-50%);
      &:hover {
        border-color: var(getThemeColor('buttonHoverBorder'));
        background-color: var(getThemeColor('buttonHoverBgColor'));
        color: var(getThemeColor('buttonHoverColor'));
      }
    }
    &--icon {
      margin-left: .4rem;
    }
  }
  &__wrapper {
    max-height: 725px;
    display: flex;
    white-space: nowrap;
    overflow: hidden;
    position: relative;
  }
  &__queue__wrapper {
    flex: 50% 1 1;
    max-height: 176px;
    min-width: 29%;
    padding-bottom: 70%;
    transition-duration: 300ms;
    transition-timing-function: ease-in-out;
    position: relative;
    border-left: 1px solid #fff;
  }
  &__static__tint__scene {
    width: 100%;
    height: 100%;
    overflow: hidden;
  }
  &__background__color {
    transition-duration: 300ms;
    transition-timing-function: ease-in-out;
  }
  &__prev-btn__wrapper {
    align-items: center;
    display: flex;
    position: absolute;
    left: 0;
    top: 50%;
    transform: translate(0%, -50%);
    z-index: 1000;
  }
  &__next-btn__wrapper {
    align-items: center;
    display: flex;
    position: absolute;
    right: 0;
    top: 50%;
    transform: translate(0%, -50%);
    z-index: 1000;
  }
  &__buttons {
    border: 1px solid var(getThemeColor('light'));
    cursor: pointer;
    font-size: 1em;
    outline: none;
    padding: 1em;
    visibility: visible;
    background-color: var(getThemeColor('white'));
    &--visible {
      visibility: hidden;
    }
    &:hover {
      border: 1px solid var(getThemeColor('secondary'));
    }
  }
  &__color__info {
    display: flex;
    flex-direction: column;
    padding: 1em;
    &__name {
      line-height: 1em;
      padding: 0 1em;
      text-align: left;
      font-weight: bold;
    }
    &__number {
      padding: 0 1em;
      margin-bottom: 0.5rem;
    }
  }
  &-house-shaped {
    &__name-number {
    display: flex;
    flex-direction: column-reverse;
    padding: 1em;
    }
    &__name {
      font-family: var(getVarName('typography.bodyFontFamily'));
      font-style: italic;
      line-height: 1.2rem;
    }
    &__number {
      font-size: .875rem;
    }
  }
}
