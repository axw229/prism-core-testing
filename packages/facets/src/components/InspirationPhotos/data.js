// @flow
import tinycolor from '@ctrl/tinycolor'
import { type Color } from 'src/shared/types/Colors.js.flow'
import uniqueId from 'lodash/uniqueId'

export const throttleDragTime = 5
// we need this two const to make sure cursor always point to the center of preview circle
export const activedPinsHalfWidth = 24

export const cloneColorPinsArr = (pins) => {
  return pins.map(pin => ({ ...pin }))
}

export const getRGBInitPins = (acgColors: Array<Object>) => {
  return acgColors.map<Object>((acgColor: any): Object => {
    return {
      rgbArray: [acgColor.r, acgColor.g, acgColor.b],
      rgbValue: `rgb(${acgColor.r},${acgColor.g},${acgColor.b})`,
      translateValueX: acgColor.x,
      translateValueY: acgColor.y
    }
  })
}

export const rgb2lab = (rgb: Array<any>) => {
  let r = rgb[0] / 255
  let g = rgb[1] / 255
  let b = rgb[2] / 255
  let x; let y; let z

  r = (r > 0.04045) ? Math.pow((r + 0.055) / 1.055, 2.4) : r / 12.92
  g = (g > 0.04045) ? Math.pow((g + 0.055) / 1.055, 2.4) : g / 12.92
  b = (b > 0.04045) ? Math.pow((b + 0.055) / 1.055, 2.4) : b / 12.92

  x = (r * 0.4124 + g * 0.3576 + b * 0.1805) / 0.95047
  y = (r * 0.2126 + g * 0.7152 + b * 0.0722) / 1.00000
  z = (r * 0.0193 + g * 0.1192 + b * 0.9505) / 1.08883

  x = (x > 0.008856) ? Math.pow(x, 1 / 3) : (7.787 * x) + 16 / 116
  y = (y > 0.008856) ? Math.pow(y, 1 / 3) : (7.787 * y) + 16 / 116
  z = (z > 0.008856) ? Math.pow(z, 1 / 3) : (7.787 * z) + 16 / 116

  return [(116 * y) - 16, 500 * (x - y), 200 * (y - z)]
}

export type RenderingPin = {
  isActiveFlag: boolean,
  name: string,
  colorNumber: number,
  rgbValue: string,
  pinNumber: number,
  isContentLeft: boolean,
  translateX: number,
  translateValueY: number,
  pinId: string
}

export const renderingPins = (initPins: Array<Object>, canvasOffsetWidth: number, canvasOffsetHeight: number, colors: Color[]): RenderingPin[] => {
  return getRGBInitPins(initPins).map<Object>((acgColor: any, index: number): Object => {
    const color: Color = findClosestColor(acgColor.rgbArray, colors)
    const calculateTranslateX = acgColor.translateValueX * canvasOffsetWidth
    const calculateTranslateY = acgColor.translateValueY * canvasOffsetHeight

    return {
      ...color,
      isActiveFlag: (initPins.length - 1 === index),
      rgbValue: `rgb(${color.red},${color.green},${color.blue})`,
      pinNumber: index,
      pinId: uniqueId('pin_'),
      isContentLeft: calculateTranslateX < canvasOffsetWidth / 2,
      translateX: Math.min(canvasOffsetWidth - 2 * activedPinsHalfWidth, calculateTranslateX),
      translateY: Math.min(canvasOffsetHeight - 2 * activedPinsHalfWidth, calculateTranslateY)
    }
  })
}

export const findClosestColor = ([red, green, blue]: [number, number, number], colors: Color[]): Color => {
  const currentPixelHex: string = tinycolor(`rgb (${red}, ${green}, ${blue})`).toHex()

  return colors.reduce(([closestColor, smallestDistanceSoFar], color): [Color, number] => {
    const colorDistance: number = measureDistance(currentPixelHex, tinycolor(`rgb(${color.red},${color.green},${color.blue})`).toHex())
    return colorDistance < smallestDistanceSoFar ? [color, colorDistance] : [closestColor, smallestDistanceSoFar]
  }, [colors[0], 99999])[0]
}

export const measureDistance = (color1: string, color2: string): number => {
  const position1 = getChromaLuma(color1)
  const position2 = getChromaLuma(color2)
  const redDelta = position1[0] - position2[0]
  const greenDelta = position1[1] - position2[1]
  const lumaDelta = position1[2] - position2[2]
  const distance = redDelta * redDelta + greenDelta * greenDelta + lumaDelta * lumaDelta
  return distance
}

export const getChromaLuma = (rgbString: string) => {
  const color = tinycolor(rgbString)
  const luminosity = color.getLuminance()
  const colorRgb = color.toRgb()
  const red = colorRgb.r
  const green = colorRgb.g
  const blue = colorRgb.b
  const sum = red + green + blue
  let redChroma
  let greenChroma
  if (sum > 25) {
    redChroma = red / sum
    greenChroma = green / sum
  } else {
    redChroma = 1 / 3
    greenChroma = 1 / 3
  }

  return [redChroma, greenChroma, luminosity]
}
