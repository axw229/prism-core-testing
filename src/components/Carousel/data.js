// This is temporary use for mock carousel data from api
// @flow
import type { Color } from '../types/Colors'
import * as Colors from '../../../__mocks__/data/color/Colors'
const colors = Colors.getAllColors()
const getExpertColors = (colors: Color, colorList: array) => {
  let data = []
  for (var i = 0; i < colorList.length; i = i + 3) {
    let tmp = []
    let j = 0
    while (j !== 3) {
      const color = Object.values(colors).find((color) => {
        return (colorList[i + j] === color.colorNumber)
      })
      tmp.push(color)
      j++
    }
    data.push(tmp)
  }
  return data
}
export const colorList = [
  '0071', '6039', '0053', '6232', '0041', '9171',
  '6496', '7063', '6747', '7579', '2739', '7502',
  '7701', '9174', '7636', '0063', '7006', '6808',
  '7622', '8917', '9110', '9178', '7538', '7578',
  '6732', '7000', '6797', '7623', '6039', '9179',
  '6224', '9175', '0034', '6839', '6071', '6656',
  '6774', '7675', '9101', '7568', '7048', '6044',
  '6424', '6573', '6680', '7592', '6244', '0037',
  '6265', '7613', '7577', '6557', '6169', '6720',
  '2844', '6221', '7104', '7001', '6650', '6967',
  '0007', '7076', '6222', '0064', '7030', '6263',
  '6052', '7566', '7666', '6788', '6841', '6156',
  '0077', '6258', '7647', '7612', '7697', '7057',
  '6510', '6131', '6199', '6464', '6556', '6771',
  '6921', '6859', '7018', '6963', '6711', '7036',
  '0073', '6463', '6353', '6164', '0012', '7674',
  '6840', '7602', '6109', '6988', '6768', '7006',
  '6759', '6344', '6802', '6349', '6020', '6307',
  '6545', '6417', '7592', '6026', '6117', '6467',
  '6247', '7031', '6155', '6979', '6725', '6963',
  '6479', '0071', '6249', '6607', '6107', '7083',
  '6018', '7058', '6256', '0056', '7662', '7135',
  '6265', '7071', '6010', '0015', '6113', '7604'
]

export const allCollectionsData = [
  {
    tabId: 'tab1',
    tabName: 'MOST POPULAR',
    collections: [
      {
        id: '1',
        name: '2019 Aficionado',
        img: 'https://www.sherwin-williams.com/dist/cvt/assets/images/collections/2019-shapeshifter-thumb.jpg',
        colorCollections: ['7579', '2739', '6404', '6407', '7705', '7502', '2704']
      },
      {
        id: '2',
        name: '2019 Naturalist',
        img: 'https://www.sherwin-williams.com/dist/cvt/assets/images/collections/2019-shapeshifter-thumb.jpg',
        colorCollections: ['6289', '9031', '6860', '9171', '6232', '9173', '0041']
      },
      {
        id: '3',
        name: '2019 Naturalist',
        img: 'https://www.sherwin-williams.com/dist/cvt/assets/images/collections/2019-shapeshifter-thumb.jpg',
        colorCollections: ['6289', '9031', '6860', '9171', '6232', '9173', '0041']
      },
      {
        id: '3',
        name: '2019 Naturalist',
        img: 'https://www.sherwin-williams.com/dist/cvt/assets/images/collections/2019-shapeshifter-thumb.jpg',
        colorCollections: ['6289', '9031', '6860', '9171', '6232', '9173', '0041']
      },
      {
        id: '3',
        name: '2019 Naturalist',
        img: 'https://www.sherwin-williams.com/dist/cvt/assets/images/collections/2019-shapeshifter-thumb.jpg',
        colorCollections: ['6289', '9031', '6860', '9171', '6232', '9173', '0041']
      },
      {
        id: '3',
        name: '2019 Naturalist',
        img: 'https://www.sherwin-williams.com/dist/cvt/assets/images/collections/2019-shapeshifter-thumb.jpg',
        colorCollections: ['6289', '9031', '6860', '9171', '6232', '9173', '0041']
      },
      {
        id: '3',
        name: '2019 Naturalist',
        img: 'https://www.sherwin-williams.com/dist/cvt/assets/images/collections/2019-shapeshifter-thumb.jpg',
        colorCollections: ['6289', '9031', '6860', '9171', '6232', '9173', '0041']
      },
      {
        id: '3',
        name: '2019 Naturalist',
        img: 'https://www.sherwin-williams.com/dist/cvt/assets/images/collections/2019-shapeshifter-thumb.jpg',
        colorCollections: ['6289', '9031', '6860', '9171', '6232', '9173', '0041']
      },
      {
        id: '3',
        name: '2019 Naturalist',
        img: 'https://www.sherwin-williams.com/dist/cvt/assets/images/collections/2019-shapeshifter-thumb.jpg',
        colorCollections: ['6289', '9031', '6860', '9171', '6232', '9173', '0041']
      }, {
        id: '3',
        name: '2019 Naturalist',
        img: 'https://www.sherwin-williams.com/dist/cvt/assets/images/collections/2019-shapeshifter-thumb.jpg',
        colorCollections: ['6289', '9031', '6860', '9171', '6232', '9173', '0041']
      },
      {
        id: '3',
        name: '2019 Naturalist',
        img: 'https://www.sherwin-williams.com/dist/cvt/assets/images/collections/2019-shapeshifter-thumb.jpg',
        colorCollections: ['6289', '9031', '6860', '9171', '6232', '9173', '0041']
      }
    ]
  },
  {
    tabId: 'tab2',
    tabName: 'COLOR ID',
    collections: [
      {
        id: '1',
        name: 'Free Spirit',
        img: 'https://www.sherwin-williams.com/dist/cvt/assets/images/collections/2019-shapeshifter-thumb.jpg',
        colorCollections: [
          '6060', '0059', '7625', '6404',
          '7743', '9169', '9086', '6116',
          '0079', '6049', '7735', '9006',
          '7640', '6369', '7551', '6992'
        ]
      }
    ]
  }
]
export const getColorCollectionsData = (colors: Color, allCollectionsData: any, tabId: string) => {
  let result = []
  let collection = allCollectionsData.find((data) => {
    return data.tabId === tabId
  })
  // temporarily added, following if condition should be removed once all collections data is available
  if (!collection) {
    collection = allCollectionsData.find((data) => {
      return data.tabId === 'tab1'
    })
  }
  collection && collection.collections.forEach((data) => {
    let tmp = {
      name: data.name,
      img: data.img,
      collections: []
    }
    for (let i = 0; i < data.colorCollections.length; i++) {
      const tmpColor = Object.values(colors).find((color) => {
        return color.colorNumber === data.colorCollections[i]
      })
      tmp.collections.push(tmpColor)
    }
    result.push(tmp)
  })
  return result
}

export const colorCollectionsData = getColorCollectionsData(colors, allCollectionsData, 'tab1')
export const expertColorsData = getExpertColors(colors, colorList)
export const collectionTabs = [
  { id: 'tab1', tabName: 'Most Popular' },
  { id: 'tab2', tabName: 'Color ID' },
  { id: 'tab3', tabName: 'Our Finest Whites' },
  { id: 'tab4', tabName: 'Color Forecast' },
  { id: 'tab5', tabName: 'Pottery Barn' },
  { id: 'tab6', tabName: 'West Elm' },
  { id: 'tab7', tabName: 'Lifestyle' },
  { id: 'tab8', tabName: `Kids' Colors` }
]
