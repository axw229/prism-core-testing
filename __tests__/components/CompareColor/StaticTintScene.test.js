/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import { StaticTintScene } from 'src/components/CompareColor/StaticTintScene'
import * as Colors from '__mocks__/data/color/Colors'
import { Scene } from 'src/components/CompareColor/data'
import ImagePreloader from 'src/helpers/ImagePreloader'

const color = Colors.getColor()

const getStaticTintScene = props => {
  let defaultProps = {
    color: color,
    scene: Scene
  }

  let newProps = Object.assign({}, defaultProps, props)
  return shallow(<StaticTintScene {...newProps} />)
}

describe('StaticTintScene with props', () => {
  let staticTintScene
  beforeEach(() => {
    if (!staticTintScene) {
      staticTintScene = getStaticTintScene()
    }
  })

  it('should render ImagePreloader', () => {
    expect(staticTintScene.find(ImagePreloader).exists()).toBe(true)
  })
})
