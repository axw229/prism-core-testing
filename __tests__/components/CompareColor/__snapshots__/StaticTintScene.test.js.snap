// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StaticTintScene with props should match snapshot with props 1`] = `
<ImagePreloader
  background="https://sherwin.scene7.com/is/image/sw?src=ir{swRender/hd_livingroom6_day?wid=300}&qlt=80"
  el={[Function]}
  height={792}
  interactive={false}
  preload={
    Array [
      undefined,
      undefined,
      undefined,
      undefined,
      MaskObj {
        "_blob": undefined,
        "_id": "1a",
        "_loadingPromise": Promise {},
        "_path": undefined,
      },
      MaskObj {
        "_blob": undefined,
        "_id": "2a",
        "_loadingPromise": Promise {},
        "_path": undefined,
      },
      MaskObj {
        "_blob": undefined,
        "_id": "3a",
        "_loadingPromise": Promise {},
        "_path": undefined,
      },
      undefined,
      undefined,
      undefined,
    ]
  }
  sceneId={1}
  surfaceStatus={
    Array [
      Object {
        "color": Object {
          "blue": 219,
          "brandKey": "SW",
          "colorFamilyNames": Array [
            "Red",
          ],
          "colorNumber": "6561",
          "coordinatingColors": Object {
            "coord1ColorId": 2250,
            "coord2ColorId": 1926,
            "whiteColorId": 2681,
          },
          "cssrgb": "rgb(235,209,219)",
          "description": Array [
            "Subdued",
            "Muted",
            "Brilliant",
            "Radiant",
          ],
          "green": 209,
          "hex": "#ebd1db",
          "hue": 0.935897435897436,
          "id": "2248",
          "isDark": false,
          "isExterior": true,
          "isInterior": true,
          "lightness": 0.8705882352941177,
          "lrv": 68.551,
          "name": "Teaberry",
          "red": 235,
          "rgb": 15454683,
          "saturation": 0.39393939393939387,
          "similarColors": Array [
            "2255",
            "2262",
            "2269",
            "2662",
            "2535",
            "2276",
            "2658",
            "1974",
            "2543",
            "2531",
          ],
          "storeStripLocator": "102-C1",
        },
        "id": 1,
      },
      Object {
        "color": Object {
          "blue": 219,
          "brandKey": "SW",
          "colorFamilyNames": Array [
            "Red",
          ],
          "colorNumber": "6561",
          "coordinatingColors": Object {
            "coord1ColorId": 2250,
            "coord2ColorId": 1926,
            "whiteColorId": 2681,
          },
          "cssrgb": "rgb(235,209,219)",
          "description": Array [
            "Subdued",
            "Muted",
            "Brilliant",
            "Radiant",
          ],
          "green": 209,
          "hex": "#ebd1db",
          "hue": 0.935897435897436,
          "id": "2248",
          "isDark": false,
          "isExterior": true,
          "isInterior": true,
          "lightness": 0.8705882352941177,
          "lrv": 68.551,
          "name": "Teaberry",
          "red": 235,
          "rgb": 15454683,
          "saturation": 0.39393939393939387,
          "similarColors": Array [
            "2255",
            "2262",
            "2269",
            "2662",
            "2535",
            "2276",
            "2658",
            "1974",
            "2543",
            "2531",
          ],
          "storeStripLocator": "102-C1",
        },
        "id": 2,
      },
      Object {
        "color": Object {
          "blue": 219,
          "brandKey": "SW",
          "colorFamilyNames": Array [
            "Red",
          ],
          "colorNumber": "6561",
          "coordinatingColors": Object {
            "coord1ColorId": 2250,
            "coord2ColorId": 1926,
            "whiteColorId": 2681,
          },
          "cssrgb": "rgb(235,209,219)",
          "description": Array [
            "Subdued",
            "Muted",
            "Brilliant",
            "Radiant",
          ],
          "green": 209,
          "hex": "#ebd1db",
          "hue": 0.935897435897436,
          "id": "2248",
          "isDark": false,
          "isExterior": true,
          "isInterior": true,
          "lightness": 0.8705882352941177,
          "lrv": 68.551,
          "name": "Teaberry",
          "red": 235,
          "rgb": 15454683,
          "saturation": 0.39393939393939387,
          "similarColors": Array [
            "2255",
            "2262",
            "2269",
            "2662",
            "2535",
            "2276",
            "2658",
            "1974",
            "2543",
            "2531",
          ],
          "storeStripLocator": "102-C1",
        },
        "id": 3,
      },
    ]
  }
  surfaces={
    Array [
      Object {
        "hitArea": "/prism/images/scenes/rooms/2/m1.svg",
        "id": 1,
        "mask": MaskObj {
          "_blob": undefined,
          "_id": "1a",
          "_loadingPromise": Promise {},
          "_path": undefined,
        },
        "role": "main",
      },
      Object {
        "hitArea": "/prism/images/scenes/rooms/2/m2.svg",
        "id": 2,
        "mask": MaskObj {
          "_blob": undefined,
          "_id": "2a",
          "_loadingPromise": Promise {},
          "_path": undefined,
        },
        "role": "main",
      },
      Object {
        "hitArea": "/prism/images/scenes/rooms/2/m3.svg",
        "id": 3,
        "mask": MaskObj {
          "_blob": undefined,
          "_id": "3a",
          "_loadingPromise": Promise {},
          "_path": undefined,
        },
        "role": "trim",
      },
    ]
  }
  type="rooms"
  width={1311}
/>
`;
